src/main/java/game/Game.java

```diff
@@ -0,0 +1,136 @@
+package game;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+public class Game {
+
+    // TODO: 02-03-2020 this is a PoC, should be replaced by more detailed and specific methods
+    public interface Listener {
+
+        void onMessage(String message);
+
+    }
+
+    private Map<Integer, Scene> sceneMap;
+    private Map<Integer, Actions> actionsMap;
+    private Map<String, Item> itemMap;
+    private Player player;
+    private int currentSceneId;
+
+    public Game() {
+        //NOP
+    }
+
+    public Game(String playerName, Map<Integer, Scene> sceneMap, Map<Integer, Actions> actionsMap, Map<String, Item> itemMap, int startSceneId) {
+        this(sceneMap, actionsMap, itemMap, startSceneId);
+        this.player = new Player(playerName);
+    }
+
+    public Game(Map<Integer, Scene> sceneMap, Map<Integer, Actions> actionsMap, Map<String, Item> itemMap, int startSceneId) {
+        this.sceneMap = sceneMap;
+        this.actionsMap = actionsMap;
+        this.itemMap = itemMap;
+        this.currentSceneId = startSceneId;
+    }
+
+    private transient Set<Listener> listeners = new CopyOnWriteArraySet<>();
+
+    /*
+    User Interaction Methods
+     */
+
+    // TODO: 02-03-2020 annotate 
+    public void start(Listener listener) {
+
+        subscribeListener(listener);
+
+        listeners.forEach(l->l.onMessage("You can write commands now!"));
+
+        handleCommand("inspect");
+    }
+
+    // TODO: 02-03-2020 annotate
+    public void handleCommand(String commandString) {
+        Command command;
+        try {
+            command = new Command(this, commandString);
+        } catch (Command.InvalidCommandException e) {
+            listeners.forEach(l->l.onMessage("Invalid command! " + e.getMessage()));
+            return;
+        }
+
+        /*
+        First check if player can handle this command
+        Second check if scene can handle this command (scene also checks its items)
+         */
+
+        Interactable.Callback callback =  new Interactable.Callback() {
+            @Override
+            public void onMessage(String message) {
+                listeners.forEach(listener -> listener.onMessage(message));
+            }
+        };
+
+        boolean isCommandHandled = player.onCommand(command, callback);
+
+        if (!isCommandHandled) {
+            isCommandHandled = sceneMap.get(currentSceneId).onCommand(command, callback);
+        }
+
+        if (!isCommandHandled) {
+            listeners.forEach(l->l.onMessage("Unknown command!"));
+        }
```

Comment:
These two will always be triggered simultaneously and can, therefore, be combined together.